ext/opentelemetry-ext-opentracing-shim/src/opentelemetry/ext/opentracing_shim/__init__.py

```diff
@@ -0,0 +1,289 @@
+
+        event_name = util.event_name_from_kv(key_values)
+        self._otel_span.add_event(event_name, event_timestamp, key_values)
+
+        return self
+
+    def set_baggage_item(self, key, value):
+        logger.warning(
+            "Calling unimplemented method set_baggage_item() on class %s",
+            self.__class__.__name__,
+        )
+        # TODO: Implement.
+
+    def get_baggage_item(self, key):
+        logger.warning(
+            "Calling unimplemented method get_baggage_item() on class %s",
+            self.__class__.__name__,
+        )
+        # TODO: Implement.
+
+    # TODO: Verify calls to deprecated methods `log_event()` and `log()` on
+    # base class work properly (it's probably fine because both methods call
+    # `log_kv()`).
+
+
+class ScopeWrapper(opentracing.Scope):
+    """A `ScopeWrapper` wraps the OpenTelemetry functionality related to span
+    activation/deactivation while using OpenTracing `Scope` objects for
+    presentation.
+
+    There are two ways to construct a `ScopeWrapper` object: using the `span`
+    argument and using the `span_cm` argument. One and only one of `span` or
+    `span_cm` must be specified.
+
+    When calling the initializer while passing a `SpanWrapper` object in the
+    `span` argument, the `ScopeWrapper` is initialized as a regular `Scope`
+    object.
+
+    When calling the initializer while passing a context manager *generator* in
+    the `span_cm` argument (as returned by the `use_span()` method of
+    OpenTelemetry `Tracer` objects,  the resulting `ScopeWrapper` becomes
+    usable as a context manager (using `with` statements).
+
+    It is necessary to have both ways for constructing `ScopeWrapper`
+    objects because in some cases we need to create the object from a context
+    manager, in which case our only way of retrieving a `Span` object is by
+    calling the `__enter__()` method on the context manager, which makes the
+    span active in the OpenTelemetry tracer; whereas in other cases we need to
+    accept a `SpanWrapper` object and wrap it in a `ScopeWrapper`.
+    """
+
+    def __init__(self, manager, span=None, span_cm=None):
+        super().__init__(manager, span)
+        self._span_cm = span_cm
+
+    def close(self):
+        self._span.unwrap().end()
+        # TODO: Set active span on OpenTelemetry tracer.
+        # https://github.com/open-telemetry/opentelemetry-python/issues/161#issuecomment-534136274
+
+    def __enter__(self):
+        if self._span_cm is not None:
+            otel_span = self._span_cm.__enter__()
+            self._span = SpanWrapper(
+                self._manager.tracer, otel_span.get_context(), otel_span
+            )
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        # Span._on_error(self.span, exc_type, exc_val, exc_tb)
+        # TODO: Do we want this ^ ? It exists in the overridden method on the
+        # base class in the OpenTracing API.
```

Comment:
That seems like a good idea, and seems like what OpenTracing users would expect to happen here.

Reply:
See https://github.com/open-telemetry/opentelemetry-python/pull/211#discussion_r334963213

Reply:
This is now irrelevant following my latest refactor as we no longer override `__exit__`.